File: xssfinder-java-executor/src/main/java/org/xssfinder/runner/TraversalStrategy.java
package org.xssfinder.runner;

import org.xssfinder.remote.TUntraversableException;
import org.xssfinder.remote.TraversalMode;

import java.lang.reflect.Method;

public interface TraversalStrategy {
    boolean canSatisfyMethod(Method method, TraversalMode traversalMode);
    TraversalResult traverse(Object page, Method method) throws TUntraversableException;
}

File: xssfinder-php-executor/src/XssFinder/Runner/TraversalStrategy.php
<?php

namespace XssFinder\Runner;

use ReflectionMethod;
use XssFinder\TraversalMode;
use XssFinder\TUntraversableException;

interface TraversalStrategy
{
    /**
     * @param ReflectionMethod $method
     * @param int $traversalMode
     * @return boolean
     */
    function canSatisfyMethod(ReflectionMethod $method, $traversalMode);


    /**
     * @param mixed $page
     * @param ReflectionMethod $method
     * @return TraversalResult
     *
     * @throws TUntraversableException
     */
    function traverse($page, ReflectionMethod $method);
}